src/main/java/app/AppBuilder.java

```diff
@@ -1,52 +1,133 @@
 package app;
 
-import view.gui_components.TradeSimulationFrame;
+import data_access.InMemoryStockDataAccessObject;
+import data_access.InMemoryUserDataAccessObject;
+import interface_adapter.execute_buy.ExecuteBuyController;
+import interface_adapter.execute_buy.ExecuteBuyPresenter;
+import use_case.execute_buy.ExecuteBuyInputBoundary;
+import use_case.execute_buy.ExecuteBuyInteractor;
+import utility.ServiceManager;
+import utility.ViewManager;
+import view.components.DialogComponent;
+import view.panels.DashboardPanel;
+import view.panels.LogInPanel;
+import view.panels.SignUpPanel;
+import view.panels.TradeSimulationPanel;
 
 import javax.swing.*;
 import java.awt.*;
 
 /**
- * A builder class for the application
+ * A builder class for the application.
+ * This class now works as a setup utility to add panels to ViewManager, build the main application frame, and manage the ServiceLocator.
  */
 public class AppBuilder {
     private final JPanel cardPanel = new JPanel();
     private final CardLayout cardLayout = new CardLayout();
 
+    // Internal ServiceLocator for managing controllers, interactors, DAOs, and presenters
+    private final ServiceManager serviceManager = new ServiceManager();
+
     public AppBuilder() {
         cardPanel.setLayout(cardLayout);
     }
 
     /**
-     * Add the trade simulation to the application
+     * Add the login and sign-up panels to the application
      *
      * @return the builder
      */
-    public AppBuilder addTradeSimulation() {
-        TradeSimulationFrame tradeSimulationFrame = new TradeSimulationFrame();
+    public AppBuilder addAuthenticationPanels() {
+        LogInPanel logInPanel = new LogInPanel();
+        SignUpPanel signUpPanel = new SignUpPanel();
+
+        // Register panels with ViewManager
+        ViewManager.Instance().registerComponent(logInPanel);
```

Comment:
Instead of adding componenents to `ViewManager` in `AppBuilder`, we can directly register each componenet within their constructors.

Reply:
Good thought.